### DISCUSSION TITLE
fishion - Latest completions from fish's source repository

### DISCUSSION THREAD

**Discussion by kidonng** [fish-shell/fish-shell#8490] [Show and tell]
Quoting myself from https://github.com/fish-shell/fish-shell/pull/8299#issue-999471922:

> I've been putting the git repo of fish in my `$fish_function_path` and `$fish_complete_path` for some time, to get faster bugfixes/better completions, due to the relatively long breaks during releases (especially my experiences with *cough, 3.2.0, cough*).

That works fine for myself, however it is still a risky move and I'd like a better way that can benefit others as well.

That's why I created [**fishion**](https://github.com/kidonng/fishion): a simple (currently) repository that **turns the latest completions from fish's source repository into a ready-to-install plugin.** It has the benefit of automated updates and safer than loading fish's repository directly.

A similar plugin for builtin functions is also planned. Stay tuned 📻

**Comment by faho** [comment]
Note that, while we're not breaking these things on purpose, there is no guarantee that the newest fish completions will work with older fishes.

**Reply by faho** [reply]
We do not, in general, support older fish releases. We simply do not have the developer time for it, and time spent making things work on older releases is much better spent fixing the actual issues.

**Reply by kidonng** [reply]
I'm aware of that (since similar things are noted in the PR I mentioned). But here are my thoughts:

- Unlike functions, most completions only use "simple" syntax that is unlikely to break in the foreseeable future
- Being a standalone project gives the flexibility that compatibility patches can be applied without needing to be accepted by fish itself

I'm confident that minimal work is required to keep them functioning.

**Reply by faho** [reply]
If you want to do that, have at it.

**Comment by krobelus** [comment]
The canonical solution would be  `make install` (perhaps with a custom `CMAKE_INSTALL_PREFIX`)
